2024/04/script.rb

```diff
@@ -0,0 +1,157 @@
+require 'pry'
+require_relative '../../tools/parser/parser.rb'
+
+class DayFour
+
+  X_BOUNDS = 139.freeze
+  Y_BOUNDS = 139.freeze
+  PHRASE = 'XMAS'.freeze
+
+  # ..X...
+  # .SAMX.
+  # .A..A.
+  # XMAS.S
+  # .X....
+  def calculate
+    mapping = []
+    (0..Y_BOUNDS).each do |y|
+      (0..X_BOUNDS).each do |x|
+        get_xmas_coords_from(y, x).each { |coord| mapping << coord }
+      end
+    end
+
+    mapping.length
+  end
+
+  def get_xmas_coords_from(y, x)
+    return [] unless normalized_input.dig(y, x) == PHRASE[0]
+
+    result = [
+      'backwards',
+      'forwards',
+      'up_left',
+      'up',
+      'down',
+      'up_right',
+      'down_right',
+      'down_left'
+    ].each_with_object([]) do |direction, accumulator|
+      result = self.send("get_xmas_#{direction}", y, x)
+
+      if result
+        accumulator << result
+      end
+
+      accumulator
+    end
+  end
+
+  private
+
+  def check_word_from_coords(coords)
+    coords = coords.filter { |x, y| x >= 0 && y >= 0 }
```

Comment:
We need to swap these two variables. The shape of objects within `coords` is not in `x, y` format, but `y, x` (as shown below)

Reply:
This always trips me up. I want it to be cartesian but it's always going to be a bit flipped. At this point I'm just happy if I can get the row number to be the second, _y_, coordinate and the column number to be the first as _x_. So I guess in your case I would leave this and change the other bit(s).